src/themes/stories/colorShowcase.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { FC } from 'react';
+import { ColorItem } from '@storybook/addon-docs';
+import { useTheme } from '@chakra-ui/react';
+
+// Not exported form the storybook addon
+type Colors =
+  | string[]
+  | {
+      [key: string]: string;
+    };
+
+const ColorShowcase: FC = () => {
+  const theme = useTheme();
+  const colors = Object.entries(theme.colors);
+
+  return (
+    <>
+      {colors.map(([colorName, colorValues]) => {
+        const mappedValues =
+          typeof colorValues === 'object'
+            ? colorValues
+            : { [colorName]: colorValues };
+
+        return (
+          <ColorItem
+            subtitle=""
+            key={colorName}
+            title={colorName}
+            colors={mappedValues as Colors}
```

Comment:
Wouldn't the type be properly inferred here?

Reply:
it originates in the `Object.entries` here as well 
![image](https://user-images.githubusercontent.com/3371760/173066119-8e8bb5e5-8165-443c-ba60-1416b5f3ac7b.png)